### PR TITLE
chore(reth-network-p2p) support no_std

### PR DESCRIPTION
**Pull request by martinezjorge**
Towards #9478 
Stacks on top of #10011

### CODE REVIEW

File: crates/net/p2p/src/error.rs [hunk 1]
```diff
@@ -207,6 +202,18 @@ impl From<DatabaseError> for DownloadError {
...
+        Self::RequestError(error)
+    }
+}
+
+impl From<ProviderError> for DownloadError {
+    fn from(error: ProviderError) -> Self {
+        Self::Provider(error)
+    }
+}
+
```
Comment (emhane) [inline crates/net/p2p/src/error.rs:216]: same here as other pr, these don't have to be done manually, use https://jeltef.github.io/derive_more/derive_more/from.html

**Review by emhane — CHANGES_REQUESTED** [review]

File: crates/net/p2p/src/error.rs (hunk 1 above)
Comment (martinezjorge) [reply crates/net/p2p/src/error.rs:216]: Thanks for the review, this will make it much less verbose!

Comment (emhane) [reply crates/net/p2p/src/error.rs:216]: exactly! and the great thing in turn with decreased verbosity, is that it's way easier to read the code that implements the functionality that sets this program apart from others

**Comment by martinezjorge** [comment]
I originally opened this PR to stack on top of #10011 because some changes I made there caused this crate to error. However, I was able to get that fixed over there without causing issues here so I'm going to close this PR.

I may eventually have to come back here but for now I'm slightly out of scope so I"ll close this for now.

**Pull request closed without merging**